src/components/Chart/ChartContainer.js

```diff
@@ -32,12 +32,11 @@ class ChartContainer extends Component {
             || prevProps.width !== this.props.width 
             || prevProps.height !== this.props.height) {
             // console.log('ComponentDidUpdate Summary Start or End or Dataset')
-            this.drawSummaryStatCharts();
+            this.drawCharts();
```

Comment:
There's something off about this function inside of componentDidUpdate and react in general. Generally drawing components should be handled by `render()`. Calls to render are triggered by updates to state. It doesn't look like there are any `forceRender()` calls in this function but this call does update state which triggers a render. I *think* this just would need to be renamed.

The other issue is that setting state in componentDidUpdate() can be tricky because it can cause an infinite loop of rendering. Just be super careful to look out for infinite render loops.

Reply:
We are frequently setting state in componentDidUpdate() like in [MainGraph](https://github.com/HopkinsIDD/covid-dashboard-app/blob/master/src/components/Graph/MainGraph.js#L69) - is this something that should be avoided in general? I'm not sure how else we would apply this logic of setting variables to state on update.

Reply:
The naming threw me off. When I see the `drawCharts` I wasn't sure if you were force rendering.

As for using setState in componentDidUpdate:

> You may call setState() immediately in componentDidUpdate() but note that it must be wrapped in a condition like in the example above, or you’ll cause an infinite loop. It would also cause an extra re-rendering which, while not visible to the user, can affect the component performance. If you’re trying to “mirror” some state to a prop coming from above, consider using the prop directly instead. Read more about why copying props into state causes bugs.

https://reactjs.org/docs/react-component.html#componentdidupdate

`drawCharts()` doesn't look simple and there are multiple possible true conditions in componentDidUpdate. If not careful, there could be many unnecessary render calls because of the way setState could trigger componentDidUpdate